react/features/conference-messages/actions.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import {
+    HIDE_CONFERENCE_MESSAGE,
+    SET_CONFERENCE_MESSAGE
+} from './actionTypes';
+
+/**
+ * Redux action to signal that the conference message needs to be hidden.
+ *
+ * @param {?string} messageKey - The message (key) to hide.
+ * @returns {{
+ *     type: HIDE_CONFERENCE_MESSAGE,
+ *     messageKey: string
+ * }}
+ *
+ */
+export function hideConferenceMessage(messageKey: ?string) {
+    return {
+        type: HIDE_CONFERENCE_MESSAGE,
+        messageKey
+    };
+}
+
+/**
+ * Redux action to signal that the conference message needs to be set.
+ *
+ * @param {?string} messageKey - The message (key) to show.
+ * @param {?number} timeout - The timeout after which the message should
+ * disappear If omited or zero, the message is permanent.
+ * @returns {Function}
+ *
+ */
+export function setConferenceMessage(messageKey: ?string, timeout: number = 0) {
+    return (dispatch: Dispatch<*>) => {
+        dispatch({
+            type: SET_CONFERENCE_MESSAGE,
+            messageKey
+        });
+
+        if (timeout !== 0) {
+            setTimeout(() => dispatch({
+                type: HIDE_CONFERENCE_MESSAGE,
+                messageKey
+            }), timeout);
```

Comment:
I think this works well enough for now. One edge case comes to mind and I don't know if it warrants any further thought. Do you think the case could happen where the messageKey gets queued at say 10minutes in the call with a 10 second timeout, then the same message gets queued at 10 minutes and 9 seconds? Right now I can't think of that causing a problem.

Reply:
Yes, that can totally cause issues, and with this, I just realized that I did the same mistake I constantly fight against: no premature features. I removed the entire timing from the messages feature, because even if I implement it properly I can't test it as I don't use it. Someone will implement it with ease when it gets used. Thanks!